src/Transformation.tsx

```diff
@@ -73,18 +92,23 @@ function Transformation() {
       return;
     }
 
-    // TODO: get inputDataCtxt's actual data from CODAP, interpret the 
-    // transformPgrm and apply it to the data, and tell CODAP to create a 
-    // new table with the transformed data.
     console.log(`Data context to transform: ${inputDataCtxt}`);
     console.log(`Transformation type: ${transformType}`);
     console.log(`Transformation to apply:\n${transformPgrm}`);
 
     const data = await(getDataFromContext(inputDataCtxt));
 
-    // TODO: Do transformation here
-
-    const newContext = await(createDataset("Testing", data));
+    const newData = data.filter(dataItem => {
+      const dataEnv = dataItemToEnv(dataItem);
+      const result = evaluate(transformPgrm, dataEnv);
+      if (result.kind !== "Bool") {
+        setErrMsg("Expected boolean output, instead got number.");
+        return false;
```

Comment:
It would be nicer to abort here. I can only think of throwing an error and using `try ... catch`. Is there a better option?

Reply:
`try ... catch` seems fine to me, we can also wrap the `evaluate` so interpreter errors get caught as well.

Reply:
Could we turn the filter into an equivalent for loop and then use return to bail out early? I agree about try/catch for evaluate though, the parser/lever/interpreter all throw errors that could be surfaced to help us fix typos in formulas